controllers/operator/manager.go

```diff
@@ -172,11 +173,22 @@ func (m *ODLMOperator) GetCatalogSourceAndChannelFromPackage(ctx context.Context
 			if excludedCatalogSources != nil && util.Contains(excludedCatalogSources, pm.Status.CatalogSource) {
 				continue
 			}
+
 			catalogsource := &olmv1alpha1.CatalogSource{}
-			if err := m.Reader.Get(ctx, types.NamespacedName{Name: pm.Status.CatalogSource, Namespace: pm.Status.CatalogSourceNamespace}, catalogsource); err != nil {
-				klog.Warning(err)
-				continue
+			hasCatalogPermission := m.CheckResAuth(ctx, namespace, "operators.coreos.com", "catalogsources", "get")
+			if hasCatalogPermission {
+				// Fetch the CatalogSource if cluster permission allows
+				catalogsource := &olmv1alpha1.CatalogSource{}
+				if err := m.Reader.Get(ctx, types.NamespacedName{Name: pm.Status.CatalogSource, Namespace: pm.Status.CatalogSourceNamespace}, catalogsource); err != nil {
+					klog.Warning(err)
+					continue
+				}
+			} else {
+				if pm.Status.CatalogSourceNamespace != namespace {
+					continue
+				}
```

Comment:
May I ask that is this condition necessary?

Also I think that it would be better if we could include the subsequent logic `currentCatalog := CatalogSource{}` into `if hasCatalogPermission` section after Catalog is obtained.